sdk/tables/azure-data-tables/azure/data/tables/_deserialize.py

```diff
@@ -96,11 +96,26 @@ def tzname(self, dt):
 
 def _from_entity_datetime(value):
     # Cosmos returns this with a decimal point that throws an error on deserialization
-    if value[-9:] == ".0000000Z":
-        value = value[:-9] + "Z"
-    return datetime.datetime.strptime(value, "%Y-%m-%dT%H:%M:%SZ").replace(
-        tzinfo=Timezone()
-    )
+    value = value.split(".")
+    ms = ""
+    if len(value) == 2:
+        ms = value[-1].replace("Z", "")
+        if len(ms) > 6:
+            ms = ms[:6]
+        ms = ms + "Z"
+        value = ".".join([value[0], ms])
+    else:
+        value = value[0]
```

Comment:
This logic is for filtering out the .Net extra decimal place right?
If that's the case - I would add comments for it... maybe even throw it into a separate function called `clean_up_dotnet_timestamps` just for fun ;)